nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -59,6 +58,36 @@ void LatticeMotionTable::initMotionModel(
   reverse_penalty = search_info.reverse_penalty;
   current_lattice_filepath = search_info.lattice_filepath;
 
+  std::ifstream latticeFile(current_lattice_filepath);
+
+  if( !latticeFile.is_open() )
+  {
+    throw std::runtime_error("Could not open lattice file");
+  }
+
+  nlohmann::json j;
+  latticeFile >> j;
+
+  fromJsonToMetaData(j["latticeMetadata"], latticeMetadata);
+
+  float prevStartAngle = 0; 
+  std::vector<MotionPrimitive> primitives; 
+  for(auto const& primative : j["primitives"] )
+  {
+    MotionPrimitive newPrimitive; 
+    fromJsonToMotionPrimitive(primative, newPrimitive); 
+
+    if(prevStartAngle != newPrimitive.start_angle )
+    {
+      motionPrimitives.emplace_back(primitives);
+      primitives.clear(); 
+      prevStartAngle = newPrimitive.start_angle;
+    }
+    primitives.emplace_back(newPrimitive);
+  }
+  motionPrimitives.emplace_back(primitives); 
+
+
   // TODO(Matt) read in file, precompute based on orientation bins for lookup at runtime
```

Comment:
These 3 TODOs can be removed now, you implemented them.

Reply:
Done